servicetalk-http-api/src/main/java/io/servicetalk/http/api/Http2ErrorCode.java

```diff
@@ -0,0 +1,169 @@
+
+    /**
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-7">CANCEL</a>.
+     */
+    public static final Http2ErrorCode CANCEL = new Http2ErrorCode(0x8, "CANCEL");
+
+    /**
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-7">COMPRESSION_ERROR</a>.
+     */
+    public static final Http2ErrorCode COMPRESSION_ERROR = new Http2ErrorCode(0x9, "COMPRESSION_ERROR");
+
+    /**
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-7">CONNECT_ERROR</a>.
+     */
+    public static final Http2ErrorCode CONNECT_ERROR = new Http2ErrorCode(0xA, "CONNECT_ERROR");
+
+    /**
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-7">ENHANCE_YOUR_CALM</a>.
+     */
+    public static final Http2ErrorCode ENHANCE_YOUR_CALM = new Http2ErrorCode(0xB, "ENHANCE_YOUR_CALM");
+
+    /**
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-7">INADEQUATE_SECURITY</a>.
+     */
+    public static final Http2ErrorCode INADEQUATE_SECURITY = new Http2ErrorCode(0xC, "INADEQUATE_SECURITY");
+
+    /**
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-7">HTTP_1_1_REQUIRED</a>.
+     */
+    public static final Http2ErrorCode HTTP_1_1_REQUIRED = new Http2ErrorCode(0xD, "HTTP_1_1_REQUIRED");
+
+    private final int errorCode;
+    private final String name;
+    private static final Http2ErrorCode[] INT_TO_ENUM_MAP;
+    static {
+        final Http2ErrorCode[] errors = new Http2ErrorCode[] {NO_ERROR, PROTOCOL_ERROR, INTERNAL_ERROR,
+                FLOW_CONTROL_ERROR, SETTINGS_TIMEOUT, STREAM_CLOSED, FRAME_SIZE_ERROR, REFUSED_STREAM, CANCEL,
+                COMPRESSION_ERROR, CONNECT_ERROR, ENHANCE_YOUR_CALM, INADEQUATE_SECURITY, HTTP_1_1_REQUIRED};
+        final Http2ErrorCode[] map = new Http2ErrorCode[errors.length];
+        for (Http2ErrorCode error : errors) {
+            map[error.code()] = error;
+        }
+        INT_TO_ENUM_MAP = map;
+    }
+
+    private Http2ErrorCode(int errorCode, String name) {
+        this.errorCode = errorCode;
+        this.name = requireNonNull(name);
+    }
+
+    /**
+     * Get the decimal value of the <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-7">error code</a>.
+     * @return the decimal value of the <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-7">
+     * error code</a>.
+     */
+    public int code() {
+        return errorCode;
+    }
+
+    /**
+     * Get the string description of the error code.
+     * @return the string description of the error code.
+     */
+    public String name() {
+        return name;
+    }
+
+    @Override
+    public String toString() {
+        return name;
+    }
+
+    @Override
+    public int hashCode() {
+        return errorCode;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        return this == o || (o instanceof Http2ErrorCode && errorCode == ((Http2ErrorCode) o).errorCode);
```

Comment:
For the final classes, it's sligtly cheaper to use `getClass()` instead of `instanceof`. Up to you.

Reply:
Then I also have to check against `null` I'll just leave as is for now we can optimize/followup for similar `final` "enum" classes wholesale if this is found to be beneficial.